pkg/model/elbv2/load_balancer.go

```diff
@@ -0,0 +1,107 @@
+package elbv2
+
+import (
+	"sigs.k8s.io/aws-alb-ingress-controller/pkg/model/core"
+)
+
+var _ core.Resource = &LoadBalancer{}
+
+// LoadBalancer represents a ELBV2 LoadBalancer.
+type LoadBalancer struct {
+	// resource id
+	id string
+
+	// desired state of LoadBalancer
+	spec LoadBalancerSpec `json:"spec"`
+
+	// observed state of LoadBalancer
+	// +optional
+	status *LoadBalancerStatus `json:"status,omitempty"`
+}
+
+// NewLoadBalancer constructs new LoadBalancer resource.
+func NewLoadBalancer(stack core.Stack, id string, spec LoadBalancerSpec) *LoadBalancer {
+	lb := &LoadBalancer{
+		id:     id,
+		spec:   spec,
+		status: nil,
+	}
+	stack.AddResource(lb)
+	lb.registerDependencies(stack)
+	return lb
+}
+
+// ID returns resource's ID within stack.
+func (lb *LoadBalancer) ID() string {
+	return lb.id
+}
+
+func (lb *LoadBalancer) registerDependencies(stack core.Stack) {
+	for _, sgToken := range lb.spec.SecurityGroups {
+		for _, dep := range sgToken.Dependencies() {
+			stack.AddDependency(lb, dep)
+		}
+	}
+}
+
+type IPAddressType string
+
+const (
+	IPAddressTypeIPV4      IPAddressType = "ipv4"
+	IPAddressTypeDualStack               = "dualstack"
+)
+
+type LoadBalancerScheme string
+
+const (
+	LoadBalancerSchemeInternal       LoadBalancerScheme = "internal"
+	LoadBalancerSchemeInternetFacing                    = "internet-facing"
+)
+
+// Information about a subnet mapping.
+type SubnetMapping struct {
+	// [Network Load Balancers] The allocation ID of the Elastic IP address for
+	// an internet-facing load balancer.
+	AllocationID *string `json:"allocationID,omitempty"`
+
+	// [Network Load Balancers] The private IPv4 address for an internal load balancer.
+	PrivateIPv4Address *string `json:"privateIPv4Address,omitempty"`
+
+	// The ID of the subnet.
+	SubnetID *string `json:"subnetID,omitempty"`
+}
+
+// LoadBalancerSpec defines the desired state of LoadBalancer
+type LoadBalancerSpec struct {
```

Comment:
Need LoadBalancerType to differentiate between NLB and ALB

Reply:
thanks :D added type here :D